baybe/simulation/core.py

```diff
@@ -96,6 +95,34 @@ def simulate_experiment(
     """
     # TODO: Due to the "..." operator, sphinx does not render this properly. Might
     #   want to investigate in the future.
+
+    context = temporary_seed(random_seed) if random_seed is not None else nullcontext()
+    with context:
+        return _simulate_experiment_given_seed(
+            campaign,
+            lookup,
+            batch_size=batch_size,
+            n_doe_iterations=n_doe_iterations,
+            initial_data=initial_data,
+            impute_mode=impute_mode,
+            noise_percent=noise_percent,
+        )
+
+
+def _simulate_experiment_given_seed(
```

Comment:
why is this new function necessary? seems the entire code could simply be called in  `simulate_experiment` right after `with context:`

Reply:
I guess you are right. I started extracting it because I played around with decorators + typing etc, and this took me so long that I at some point forgot about the simple "with" option, which I rejected at the beginning because of that trivial and large increased indent spanning the entire function body, which somehow cried for another solution. But I agree, having the new function header is definitely worse than the indent. Will change and then we can merge

Reply:
changed